### ISSUE TITLE
PHP 8.1: calling Url::short('') throws deprecation warning

### ISSUE THREAD

**Issue by mrflix** [k-next/kirby#6121]
## Description
I'm calling `URL::short` and in some cases my url is null. In those cases I get the PHP 8.1 deprecation warning.

**Expected behavior**  
I'd expect Kirby to prevent it. But I'm not 100% sure. The simple `$value ?? ''` doesn't work here because `URL::short` sets the url to null if its an empty string:

```php
/**
 * Shortens the Url by removing all unnecessary parts
 */
public static function short(
	string|null $url = null,
	int $length = 0,
	bool $base = false,
	string $rep = '…'
): string {
	$uri = static::toObject($url);

	$uri->fragment = null;
	$uri->query    = null;
	$uri->password = null;
	$uri->port     = null;
	$uri->scheme   = null;
	$uri->username = null;

	// remove the trailing slash from the path
	$uri->slash = false;

	$url = $base ? $uri->base() : $uri->toString();
	$url = str_replace('www.', '', $url);

	return Str::short($url, $length, $rep);
}
```

## To reproduce
Call `URL::short('')` with PHP 8.1 or higher.

## Your setup
**Kirby Version**  
4.0.2

**Comment by distantnative** [comment]
Running PHP 8.2 and not getting any warnings, neither calling it with `null` nor with `''`

**Comment by afbora** [comment]
I can't reproduce the issue on PHP 8.1 either. `Url::short()` covers `null` and empty values.

**Comment by mrflix** [comment]
Nico and Ahmet thanks for taking the time. I now found out why it's not reproducible: it only happens with the third parameter set to true.

```php
Url::short('', 0, true);
```

This parameter is called 'base' and it makes sure to only return the base "github.com" instead of the whole path "github.com/getkirby/kirby/issues/6121".

**Comment by bastianallgeier** [comment]
I think I have a fix: https://github.com/getkirby/kirby/pull/6126

**Comment by mrflix** [comment]
That fix (?? '') solves 90% of my 8.1 errors 😬
